repos/system_upgrade/common/actors/opensshconfigscanner/tests/test_readopensshconfig_opensshconfigscanner.py

```diff
@@ -143,12 +150,66 @@ def test_parse_config_deprecated():
 def test_parse_config_empty():
     output = parse_config([])
     assert isinstance(output, OpenSshConfig)
-    assert isinstance(output, OpenSshConfig)
     assert not output.permit_root_login
     assert output.use_privilege_separation is None
     assert output.protocol is None
 
 
+def test_parse_config_include():
```

Comment:
Instead of using temporary files, you abstract away lines where you are reading from disk in the scanner, e.g.,
```python3
       with open(config, 'r') as fd:
            return fd.readlines()
```
would become a new function within th e actor. Then you can mock this function within your tests, avoiding interacting with filesystem altogether.

Reply:
Given that the actor goes over several files that are included from each other, I found it more useful to have the real directory structure instead of mocking the reading of different files. I would need to mock also the globing, which felt I would have to replace most of the logic in there.

Reply:
@Jakuje Out of curiosity I have tried to rewrite the test without using filesystem https://github.com/MichalHe/leapp-repository/commit/a4d496f8c89d1a04e63f8042cdb9d9ed2bd8d2eb. What do you think about the approach? If you are adamant about using temporary files, then we can close this conversation and let this patch be merged with temporary files. I do not like relying on temporary files if it can be avoided in a sane fashion because the less we rely on anything the less things can go wrong. Maybe I am just paranoid because upgrades thought me not to trust any aspect of the system I am running on :grin: 

If you consider the approach proposed by me OK, then feel free to just cherry-pick the commit from my branch.

Reply:
sure. I will use your proposal. It looks cleaner. I got almost that far to implement it when I realized the glob needs to be mocked too and gave up.